.github/workflows/release.yml

```
@@ -0,0 +1,19 @@
+---
+name: "Release"
+on:
+  push:
+    tags:
+      - 'v*.*.*'
+
+jobs:
+  luarocks-release:
+    runs-on: ubuntu-latest
+    name: Luarocks Release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```

```suggestion
        uses: actions/checkout@v3
        if: env.LUAROCKS_API_KEY != null
```

Would avoid the checkout costs if the key is not set.